FinSearchUnified/Tests/Subscriber/FrontendTest.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+use Enlight_Controller_Request_RequestHttp as RequestHttp;
+use FinSearchUnified\finSearchUnified as Plugin;
+use Shopware\Components\Test\Plugin\TestCase;
+
+class FrontendTest extends TestCase
+{
+    /**
+     * @return array
+     */
+    public function frontendPreDispatchProvider()
+    {
+        return [
+            'Search Page' => [
+                'sSearch' => 'Yes',
+                'sCategory' => null,
+                'sController' => 'index',
+                'sAction' => 'index',
+                'sModule' => null
+            ],
+            'Category Page' => [
+                'sSearch' => null,
+                'sCategory' => 3,
+                'sController' => 'listing',
+                'sAction' => 'index',
+                'sModule' => null
+            ],
+            'Manufacturer Page' => [
+                'sSearch' => null,
+                'sCategory' => null,
+                'sController' => 'listing',
+                'sAction' => 'manufacturer',
+                'sModule' => null
+            ],
+            'Backend Module' => [
+                'sSearch' => null,
+                'sCategory' => null,
+                'sController' => 'index',
+                'sAction' => 'index',
+                'sModule' => 'backend'
+            ],
+            'Current Page is not Listing or Search' => [
+                'sSearch' => null,
+                'sCategory' => null,
+                'sController' => 'index',
+                'sAction' => 'index',
+                'sModule' => null
+            ]
+        ];
+    }
+
+    public function listingCountConditionProvider()
+    {
+        return [
+            'Check values after listingCount call on Search Page' => [
+                'sSearch' => 'Yes',
+                'sCategory' => null,
+                'sController' => 'listing',
+                'sAction' => 'listingCount',
+                'sModule' => 'widgets'
+            ],
+            'Check values after listingCount call in Category Page' => [
+                'sSearch' => null,
+                'sCategory' => 3,
+                'sController' => 'listing',
+                'sAction' => 'listingCount',
+                'sModule' => 'widgets'
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider frontendPreDispatchProvider
+     *
+     * @param string $sSearch
+     * @param int|null $sCategory
+     * @param string $sController
+     * @param string $sAction
+     * @param string $sModule
+     */
+    public function testFrontendPreDispatchConditions($sSearch, $sCategory, $sController, $sAction, $sModule)
```

Comment:
I could be wrong but it seems that this method isn't actually testing anything. I can't see any assertions at least.

Reply:
ah yes, my bad. I think I forgot to add the assertions after we refactored a separate test method for the listingCount scenario.